admin/RequestSolrCloudBackups

```diff
@@ -0,0 +1,308 @@
+#    running     => 'Solr Cloud is processing backup request',
+#    completed   => 'Solr Cloud completely processed backup request',
+#);
+
+my $collections = $sample_collection
+    ? [$sample_collection]
+    : list_solr_collections();
+my %backup_requests;
+prepare_backup_requests();
+do_backup_collections();
+wait_backup_complete();
+_info 'Successfully completed backup for all collections on SolrCloud.';
+exit;
+
+################################################################################
+
+sub list_solr_collections
+{
+    my $solr_response = _query_collections_api('LIST');
+    return $solr_response->{collections};
+}
+
+sub prepare_backup_requests
+{
+    my $next_request_id = 1500;
+    my $has_error;
+    foreach my $collection (sort @$collections)
+    {
+        my $request_id = $next_request_id++;
+        $backup_requests{$request_id} = {
+            name => 'backup_' . $BACKUP_STAMP . '_' . $collection,
+            collection => $collection,
+            status => 'pending',
+        };
+        my $solr_response = _query_collections_api('REQUESTSTATUS', {
+                requestid => $request_id,
+            }
+        );
+        my $solr_request_status = $solr_response->{status}->{state};
+        if ($solr_request_status ne 'notfound')
+        {
+            _error(
+                "A request to SolrCloud with id '%d' is '%s' already: '%s'",
+                $request_id,
+                $solr_request_status,
+                encode_json($solr_response),
+            );
+            $has_error = 1;
+        }
+    }
+    exit 70 if $has_error; # EX_SOFTWARE
+}
+
+sub do_backup_collections
+{
+    foreach my $request_id (sort keys %backup_requests)
+    {
+        my $backup_name = $backup_requests{$request_id}{name};
+        my $collection = $backup_requests{$request_id}{collection};
+
+        _debug (
+            "Sending backup request '%d' of collection '%s' as '%s'.",
+            $request_id,
+            $collection,
+            $backup_name,
+        );
+        my $solr_response = _query_collections_api('BACKUP', {
+                name => $backup_name,
+                collection => $collection,
+                location => $BACKUP_LOCATION,
+                async => $request_id,
+            }
+        );
+        if (defined $solr_response->{error})
+        {
+            _error(
+                "Failed to backup request '%s' of collection '%s' as '%s': " .
+                "Error '%s'",
+                $request_id,
+                $collection,
+                $backup_name,
+                $solr_response->{error},
+            );
+            _flush_running_requests_status();
+            exit 70; # EX_SOFTWARE
+        }
+        $backup_requests{$request_id}{status} = 'running';
+    }
+}
+
+sub wait_backup_complete
+{
+    while (
+        sum
+        map { $_ => 1 }
+        grep { $backup_requests{$_}{status} eq 'running' }
+        keys %backup_requests
```

Comment:
Not sure I understand what this is doing. `sum map { $_ => 1 }` will sum all the keys and values together, but what does that mean?

Reply:
It just counts backup requests that are still running.

Reply:
Oh. That makes sense now, but just `while (grep { $backup_requests{$_}{status} eq 'running' } keys %backup_requests)` should suffice unless I'm missing something.

Reply:
Indeed, removed count in https://github.com/metabrainz/musicbrainz-server/pull/1407/commits/728b7f7945abe38ffa1d55cd372c8aa5157c8341.